edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/CloudConnection.cs

```diff
@@ -263,33 +264,55 @@ void InternalConnectionStatusChangesHandler(ConnectionStatus status, ConnectionS
         async Task<string> GetNewToken(string iotHub, string id, string currentToken, IIdentity currentIdentity)
```

Comment:
Could you please add some comments explaining the retry logic.

Reply:
Sure.